src/git-repository-async.js

```diff
@@ -0,0 +1,271 @@
+'use babel'
+
+const Git = require('nodegit')
+const path = require('path')
+const {Emitter, Disposable, CompositeDisposable} = require('event-kit')
+
+// Temporary requires
+// ==================
+// GitUtils is temporarily used for ::relativize only, because I don't want
+// to port it just yet. TODO: remove
+const GitUtils = require('git-utils')
+// Just using this for _.isEqual and _.object, we should impl our own here
+const _ = require('underscore-plus')
+
+module.exports = class GitRepositoryAsync {
+  static open (path, options = {}) {
+    // QUESTION: Should this wrap Git.Repository and reject with a nicer message?
+    return new GitRepositoryAsync(path, options)
+  }
+
+  constructor (path, options) {
+    this.repo = null
+    this.emitter = new Emitter()
+    this.subscriptions = new CompositeDisposable()
+    this.pathStatusCache = {}
+    this._gitUtilsRepo = GitUtils.open(path) // TODO remove after porting ::relativize
+    this.repoPromise = Git.Repository.open(path)
+
+    var {project, refreshOnWindowFocus} = options
+    this.project = project
+    if (refreshOnWindowFocus === undefined) {
+      refreshOnWindowFocus = true
+    }
+    if (refreshOnWindowFocus) {
+      // TODO
+    }
+
+    if (this.project) {
+      this.subscriptions.add(this.project.onDidAddBuffer((buffer) => {
+        this.subscribeToBuffer(buffer)
+      }))
+
+      this.project.getBuffers().forEach((buffer) => { this.subscribeToBuffer(buffer) })
+    }
+  }
+
+  destroy () {
+    this.subscriptions.dispose()
+  }
+
+  getPath () {
+    return this.repoPromise.then((repo) => {
+      return Promise.resolve(repo.path().replace(/\/$/, ''))
+    })
+  }
+
+  isPathIgnored (_path) {
+    return this.repoPromise.then((repo) => {
+      return Promise.resolve(Git.Ignore.pathIsIgnored(repo, _path))
+    })
+  }
+
+  isPathModified (_path) {
+    return this._filterStatusesByPath(_path).then(function (statuses) {
+      return statuses.filter((status) => {
+        return status.isModified()
```

Comment:
This can be changed into
```js
return statuses.filter(status => status.isModified())
```

You should probably also be able to rewrite the whole block as
```js
return this._filterStatusesByPath(_path).then(statuses => statuses.filter(status => status.isModified()))
```